src/raps/utils.ts

```diff
@@ -0,0 +1,48 @@
+import { Network } from '@/helpers';
+import {
+  GasFeeParamsBySpeed,
+  LegacyGasFeeParamsBySpeed,
+  LegacyTransactionGasParamAmounts,
+  TransactionGasParamAmounts,
+} from '@/entities';
+import { ethereumUtils, gasUtils } from '@/utils';
+import { add, greaterThan } from '@/helpers/utilities';
+
+export const overrideWithFastSpeedIfNeeded = ({
+  gasParams,
+  chainId,
+  gasFeeParamsBySpeed,
+}: {
+  gasParams: TransactionGasParamAmounts | LegacyTransactionGasParamAmounts;
+  chainId: number;
+  gasFeeParamsBySpeed: GasFeeParamsBySpeed | LegacyGasFeeParamsBySpeed;
+}): TransactionGasParamAmounts | LegacyTransactionGasParamAmounts => {
+  if (chainId !== ethereumUtils.getChainIdFromNetwork(Network.mainnet)) {
```

Comment:
so we only want to do this for mainnet or for all 1559 chains?

Reply:
only mainnet - referring to the convo with Wayne about how this is less necessary on L2s in terms of risk of expiry for the following swaps (and lines up more with BX)